src/forms/TogglePill.jsx

```diff
@@ -1,26 +1,25 @@
 import React from 'react';
 import cx from 'classnames';
 import Icon from '../media/Icon';
+import withToggleControl from '../utils/WithToggleControl';
 
 export const TOGGLE_PILL_CLASS = 'toggleButton';
 
 /**
  * SQ2 Toggle Pill component
  * @see {@link https://github.com/meetup/sassquatch2/blob/develop/sass/ui-components/_toggle-pill.scss}
  * @see {@link http://meetup.github.io/sassquatch2/ui_components.html#togglePills}
- * @module TogglePill
+ * @module TogglePillBase
  */
-export default class TogglePill extends React.Component {
+class TogglePillBase extends React.Component {
 	constructor (props) {
 		super(props);
 
-		this.state = {isChecked: props.checked || false};
-
 		this.onChange = this.onChange.bind(this);
 	}
 
 	onChange(e) {
-		this.setState({isChecked: !this.state.isChecked});
+		this.props.toggle();
```

Comment:
Seeing the `toggle` prop in use here makes me want to suggest a clearer name, like `toggleActive`. Because `withToggleControl` doesn't know what prop name will be defined, I know this makes it more difficult to come up with a clear name here. 

Supporting a custom "checked" prop name is a cool feature, but I'm now wondering what is gained. If we have a number of components using `withToggleControl`, it may be a disadvantage if each component has a different prop name to deal with the same behavior. 

Forcing the name of the prop to always be the same might make it easier for developers to understand what's going on under the hood (and to remember the interface for any components using `withToggleControl`. Because not all toggles will use checkboxes, mabye:

- `isActive` for the prop name
- `state.isActive` instead of `state.bool`
- `toggleActive` instead of `toggle`

Setting a checkbox `checked` attribute with `this.props.isActive` instead of `this.props.isChecked` seems like a small price to pay for a more consistent interface